### PR TITLE
Utah taxpayer credit calculations

### LINKED ISSUE #3553: Fix Utah taxpayer credit calculations
In the 0.601.0 version, there are at least two problems with the Utah taxpayer credit logic.

First, the `/ut/tax/income/credits/taxpayer/refundable.yaml` parameter file is redundant, and therefore, error-prone.  It should be eliminated.

Second, the formula for the `ut_federal_deductions_for_taxpayer_credit` variable is incorrect.  The instructions make it clear that not the whole federal SALT deduction is entered on Line 14.   The taxpayer credit is calculated on an amount that includes (capped) real-estate (that is, local property) taxes.

Screenshot 2023-12-30 at 12 52 19 PM

### PR DESCRIPTION
**Pull request by JudyZ13**
Fixes #3553

### CODE REVIEW

File: changelog_entry.yaml [hunk 1]
```diff
@@ -0,0 +1,4 @@
+- bump: minor
+  changes:
+    added:
```
Comment (PavelMakarchuk) [inline changelog_entry.yaml:3]:
```suggestion
    fixed:
```

**Review by PavelMakarchuk — CHANGES_REQUESTED** [review]

File: policyengine_us/variables/gov/states/oh/tax/income/exemptions/oh_personal_exemptions.py [hunk 2]
```diff
@@ -20,6 +20,6 @@ def formula(tax_unit, period, parameters):
         )
 
         agi = tax_unit("oh_agi", period)
-        exemption_amount = p.amount.calc(agi, right=True)
+        credit_amount = p.amount.calc(agi, right=True)
```
Comment (PavelMakarchuk) [inline policyengine_us/variables/gov/states/oh/tax/income/exemptions/oh_personal_exemptions.py:23]:
```suggestion
        exemption_amount = p.amount.calc(agi, right=True)
```

File: policyengine_us/variables/gov/states/oh/tax/income/exemptions/oh_personal_exemptions.py [hunk 3]
```diff
@@ -20,6 +20,6 @@ def formula(tax_unit, period, parameters):
...
 
-        return eligible_exemptions * exemption_amount
+        return eligible_exemptions * credit_amount
```
Comment (PavelMakarchuk) [inline policyengine_us/variables/gov/states/oh/tax/income/exemptions/oh_personal_exemptions.py:25]:
```suggestion
        return eligible_exemptions * exemption_amount
```

File: policyengine_us/tests/policy/baseline/gov/states/ut/tax/income/credits/ut_federal_deductions_for_taxpayer_credit.yaml
Comment (PavelMakarchuk) [inline policyengine_us/tests/policy/baseline/gov/states/ut/tax/income/credits/ut_federal_deductions_for_taxpayer_credit.yaml:1]: give tests unique names

File: policyengine_us/variables/gov/states/ut/tax/income/credits/taxpayer_credit/ut_federal_deductions_for_taxpayer_credit.py [hunk 4]
```diff
@@ -12,7 +12,20 @@ class ut_federal_deductions_for_taxpayer_credit(Variable):
     reference = "https://le.utah.gov/xcode/Title59/Chapter10/59-10-S114.html?v=C59-10-S114_2022032320220323"
 
     def formula(tax_unit, period, parameters):
-        us_itemizing = tax_unit("tax_unit_itemizes", period)
-        itm_ded_less_salt = tax_unit("itemized_deductions_less_salt", period)
-        std_ded = tax_unit("standard_deduction", period)
```
Comment (PavelMakarchuk) [inline policyengine_us/variables/gov/states/ut/tax/income/credits/taxpayer_credit/ut_federal_deductions_for_taxpayer_credit.py:17]: we never added back the standard deduction here

File: policyengine_us/variables/gov/states/ut/tax/income/credits/taxpayer_credit/ut_federal_deductions_for_taxpayer_credit.py [hunk 5]
```diff
@@ -12,7 +12,20 @@ class ut_federal_deductions_for_taxpayer_credit(Variable):
...
-        return where(us_itemizing, itm_ded_less_salt, std_ded)
+        p = parameters(period).gov.irs.deductions
```
Comment (PavelMakarchuk) [inline policyengine_us/variables/gov/states/ut/tax/income/credits/taxpayer_credit/ut_federal_deductions_for_taxpayer_credit.py:15]: add comments for each line

Comment (PavelMakarchuk) [reply policyengine_us/variables/gov/states/ut/tax/income/credits/taxpayer_credit/ut_federal_deductions_for_taxpayer_credit.py:16]: The following states that the filer can pick either the standard or the itemized deduction
you need 
```
        std_ded = tax_unit("standard_deduction", period)
        us_itemizing = tax_unit("tax_unit_itemizes", period)
```


image

Comment (PavelMakarchuk) [reply policyengine_us/variables/gov/states/ut/tax/income/credits/taxpayer_credit/ut_federal_deductions_for_taxpayer_credit.py:16]: please in the code, add line comments from this source

File: policyengine_us/variables/gov/states/ut/tax/income/credits/taxpayer_credit/ut_federal_deductions_for_taxpayer_credit.py [hunk 6]
```diff
@@ -12,7 +12,23 @@ class ut_federal_deductions_for_taxpayer_credit(Variable):
     reference = "https://le.utah.gov/xcode/Title59/Chapter10/59-10-S114.html?v=C59-10-S114_2022032320220323"
 
     def formula(tax_unit, period, parameters):
-        us_itemizing = tax_unit("tax_unit_itemizes", period)
-        itm_ded_less_salt = tax_unit("itemized_deductions_less_salt", period)
-        std_ded = tax_unit("standard_deduction", period)
-        return where(us_itemizing, itm_ded_less_salt, std_ded)
+        # itemized deductions (exclude SALT)
+        p = parameters(period).gov.irs.deductions
+
```
Comment (PavelMakarchuk) [inline policyengine_us/variables/gov/states/ut/tax/income/credits/taxpayer_credit/ut_federal_deductions_for_taxpayer_credit.py:17]:
```suggestion
     # Line 12 standard or itemized deduction
     st_ded = tax_unit...
     .
     .
     .
     # Line 13
```

File: policyengine_us/variables/gov/states/ut/tax/income/credits/taxpayer_credit/ut_federal_deductions_for_taxpayer_credit.py [hunk 7]
```diff
@@ -12,7 +12,26 @@ class ut_federal_deductions_for_taxpayer_credit(Variable):
...
         std_ded = tax_unit("standard_deduction", period)
-        return where(us_itemizing, itm_ded_less_salt, std_ded)
+
+        # Subtract SALT from Itemized Deductions
+        deductions = [
+            deduction
+            for deduction in p.itemized_deductions
+            if deduction not in ["salt_deduction"]
+        ]
+        ite_ded = add(tax_unit, period, deductions)
```
Comment (PavelMakarchuk) [inline policyengine_us/variables/gov/states/ut/tax/income/credits/taxpayer_credit/ut_federal_deductions_for_taxpayer_credit.py:27]:
```suggestion
        item_ded = add(tax_unit, period, deductions)
```

File: policyengine_us/variables/gov/states/ut/tax/income/credits/taxpayer_credit/ut_federal_deductions_for_taxpayer_credit.py [hunk 8]
```diff
@@ -12,7 +12,26 @@ class ut_federal_deductions_for_taxpayer_credit(Variable):
...
+
+        # Include Real Estate Taxes in Itemized Deductions
+        real_estate_tax = add(tax_unit, period, ["real_estate_taxes"])
+
+        capped_real_estate_tax = min_(
+            real_estate_tax, p.itemized.salt_and_real_estate.cap[filing_status]
+        )
+
+        # Line 12. Federal Standard or Itemized Deductions
+        return where(us_itemizing, ite_ded + capped_real_estate_tax, std_ded)
```
Comment (PavelMakarchuk) [inline policyengine_us/variables/gov/states/ut/tax/income/credits/taxpayer_credit/ut_federal_deductions_for_taxpayer_credit.py:37]:
```suggestion
        total_item_ded = item_ded + capped_real_estate_tax
        return where(us_itemizing, total_item_ded, std_ded)
```

**Review by PavelMakarchuk — CHANGES_REQUESTED** [review]

**Review by MaxGhenis — APPROVED** [review]

**Pull request merged**